winml/lib/Api.Image/D3DDeviceCache.cpp

```diff
@@ -49,37 +49,48 @@ D3DDeviceCache::D3DDeviceCache(winml::LearningModelDeviceKind const& deviceKind)
     return;
   }
 
-  DXGI_GPU_PREFERENCE preference;
-  WINML_THROW_IF_FAILED(GetGPUPreference(deviceKind, &preference));
-
   CommonDeviceHelpers::AdapterEnumerationSupport support;
   WINML_THROW_IF_FAILED(CommonDeviceHelpers::GetAdapterEnumerationSupport(&support));
 
   const char noHardwareAdaptersAvailableErrStr[] = "No hardware adapters available";
   const char failedToObtainHardwareAdaptersErrStr[] = "Failed to obtain hardware adapters.";
   HRESULT hardwareAdapterSuccessfullyObtained = S_OK;
-  if (support.has_dxgi) {
-    winrt::com_ptr<IDXGIAdapter1> spAdapter;
-    hardwareAdapterSuccessfullyObtained = GetDXGIHardwareAdapterWithPreference(preference, spAdapter.put());
+
+  if (support.has_dxcore && deviceKind == winml::LearningModelDeviceKind::ComputeAccelerator) {
+    winrt::com_ptr<IDXCoreAdapter> spAdapter;
+    hardwareAdapterSuccessfullyObtained = GetDXCoreHardwareAdapterForVPU(spAdapter.put());
     if (hardwareAdapterSuccessfullyObtained == HRESULT_FROM_WIN32(ERROR_NOT_FOUND)) {
```

Comment:
>HRESULT_FROM_WIN32(ERROR_NOT_FOUND) [](http://example.com/codeflow?start=47&length=35)

Are you sure this is the specific error code that DXCore returns in the case that enumeration succeeds, but no adapters are found? If checking against a specific error code is desired, perhaps `GetDXCoreHardwareAdapterForVPU` should should instead explicitly check the adapter count and return `HRESULT_FROM_WIN32(ERROR_NOT_FOUND)`, instead of implicitly relying on DXCore to do that in the cases you want.